src/mem/arena.rs

```diff
@@ -32,6 +32,18 @@ use crate::mem::align_to;
 #[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub struct OutOfMemory;
 
+impl OutOfMemory {
+    /// Returns a static, mutable reference to an `OutOfMemory`.
+    ///
+    /// This function is the moral equivalent of `&mut []`; `OutOfMemory`
+    /// holds no data, so any non-null pointer is a suitable reference to
+    /// `OutOfMemory`.
```

Comment:
This comment is technically incorrect (the pointer also needs to be aligned, and the alignment isn't *guaranteed* to be 1). It can probably be shortened to "This function is similar to `&mut []`, as `OutOfMemory` holds no data.".